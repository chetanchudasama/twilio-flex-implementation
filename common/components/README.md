# Common Components Library

The common components library contains UI elements and models that can be used across multiple Flex plugins.

It is written in react v16.13.1 and typescript v3.

## Contents

## Pre-requisites

See Pre-requisites section in the [main README.md file](../../README.md).

## Building the library

See Scripts section in the [main README.md file](../../README.md) for building and running the unit tests for this library.
When building the library, the components contained within will be compiled to CommonJS and ESM formats and a `lib` folder will be created containing the compiled files. The Flex plugins will then be able to reference any of the components in the library.
