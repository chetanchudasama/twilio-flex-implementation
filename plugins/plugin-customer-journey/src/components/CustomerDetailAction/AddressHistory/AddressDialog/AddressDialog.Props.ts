import { AddressItemModel } from "@common/components";

export interface AddressDialogProps {
  isEdit?: boolean;
  open: boolean;
  loading: boolean;
  onDelete: () => void;
  onSave: (address: AddressItemModel) => void;
  onClose: () => void;
  activeAddress: AddressItemModel;
}
