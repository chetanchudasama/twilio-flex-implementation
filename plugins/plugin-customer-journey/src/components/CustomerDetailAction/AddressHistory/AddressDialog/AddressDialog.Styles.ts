import { paddingUnit, errorColor } from "@common/components";
import { Grid } from "@material-ui/core";
import { styled } from "@twilio/flex-ui";

export const AddressDialogStyles = styled(Grid)({});

export const ResponsiveDialogContentStyles = styled(Grid)({
  padding: paddingUnit * 2,
  ".address-container > div": {
    padding: 0,
  },
  ".address-container .address-configuration": {
    paddingTop: 2,
  },
  ".address-container .address-current": {
    marginTop: 2,
  },
  ".address-error": {
    fontSize: 12,
    color: errorColor,
    marginTop: 4,
  },
});
