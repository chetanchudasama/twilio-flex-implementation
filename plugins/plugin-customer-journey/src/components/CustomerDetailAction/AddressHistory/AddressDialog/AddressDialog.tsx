import {
  AddressItemModel,
  CustomSelect,
  ResponsiveDialog,
  Shared,
  BaseAddressModel,
  ResidentialStatusModel,
} from "@common/components";
import {
  Checkbox,
  FormControlLabel,
  Grid,
  MenuItem,
  InputLabel,
} from "@material-ui/core";
import React, { useMemo, useState } from "react";
import { v4 } from "uuid";
import { AddressDialogProps } from "./AddressDialog.Props";
import {
  AddressDialogStyles,
  ResponsiveDialogContentStyles,
} from "./AddressDialog.Styles";
import Loading from "../../../LoadingComponent/Loading";
import { AddressSearch } from "../../../AddressSearch/AddressSearch";

export const AddressDialog: React.FC<AddressDialogProps> = ({
  activeAddress,
  open,
  onClose,
  onDelete,
  onSave,
  loading,
  isEdit,
}) => {
  const [openConfirmation, setOpenConfirmation] = useState<boolean>(false);
  const [errors, setErrors] = useState<Record<string, string[]>>({});
  const [address, setAddress] = useState<AddressItemModel>(activeAddress);

  const isAddressExist: boolean = useMemo(() => {
    return !!(
      address.postcode ||
      address.buildingName ||
      address.subBuilding ||
      address.buildingNumber ||
      address.streetName ||
      address.town
    );
  }, [address]);

  const onChangeAddressDetail = (event: any, propertyName: string) => {
    const {
      target: { value, checked },
    } = event;

    if (propertyName === "isPrimaryAddress") {
      setAddress({
        ...address,
        [propertyName]: checked,
      });
      return;
    }
    if (propertyName === "residentialStatus") {
      const residentialStatus: ResidentialStatusModel | undefined =
        Shared.residentialStatuses.find(
          (s: ResidentialStatusModel) => s.id === Number(value)
        );
      setAddress({
        ...address,
        addressStatus: {
          ...address.addressStatus,
          addressStatusId: residentialStatus ? residentialStatus.id : -1,
          addressStatusName: residentialStatus ? residentialStatus.name : "",
        },
      });
    } else {
      setAddress({
        ...address,
        addressStatus: {
          ...address.addressStatus,
          [propertyName]: value,
        },
      });
    }
  };

  const onSaveAddressDetail = () => {
    const errorObj: Record<string, string[]> = {};
    if (address.addressStatus.addressStatusId <= 0) {
      errorObj.residentialStatus = ["The residential status is required."];
    }

    if (
      address.addressStatus.yearsAtAddress === 0 &&
      address.addressStatus.monthsAtAddress === 0
    ) {
      errorObj.residentialPeriod = ["Residential period is required."];
    }

    if (!isAddressExist) {
      errorObj.customerAddress = ["Address of customer is required."];
    }

    setErrors(errorObj);

    if (Object.keys(errorObj).length === 0) {
      onSave(address);
    }
  };

  const handleUpdateAddress = (baseAddressModel: BaseAddressModel) => {
    setAddress({
      ...address,
      buildingName: baseAddressModel.buildingName,
      buildingNumber: baseAddressModel.buildingNumber,
      postcode: baseAddressModel.postcode,
      streetName: baseAddressModel.streetName,
      subBuilding: baseAddressModel.subBuilding,
      town: baseAddressModel.town,
    });
  };

  return (
    <AddressDialogStyles item xs={12}>
      {open && (
        <ResponsiveDialog
          title={isEdit ? "Edit Address" : "Add Address"}
          open={open}
          onCancel={onClose}
          onConfirm={onSaveAddressDetail}
          maxWidth="sm"
          showActionButtons
          showDeleteButton={isEdit}
          onDelete={() => {
            setOpenConfirmation(true);
          }}
          okText="Save Address"
          cancelText="Cancel"
          deleteText="Delete Address"
        >
          <ResponsiveDialogContentStyles item xs={12}>
            <Grid
              container
              direction="row"
              justify="space-between"
              alignItems="center"
              spacing={8}
            >
              <Grid item xs={12} className="address-container">
                <InputLabel shrink required error={!!errors.customerAddress}>
                  Address
                </InputLabel>
                <AddressSearch
                  updateAddress={handleUpdateAddress}
                  address={isAddressExist ? activeAddress : undefined}
                />
                {!!errors.customerAddress && (
                  <p className="address-error">{errors.customerAddress}</p>
                )}
              </Grid>
              <Grid item xs={12} className="residential-status">
                <CustomSelect
                  label="Residential status"
                  onChange={(event: any) =>
                    onChangeAddressDetail(event, "residentialStatus")
                  }
                  value={address.addressStatus.addressStatusId || -1}
                  isRequired
                  errors={errors.residentialStatus}
                >
                  <MenuItem value={-1}>
                    <em>Select a status</em>
                  </MenuItem>
                  {Shared.residentialStatuses.map(
                    (status: ResidentialStatusModel) => (
                      <MenuItem key={v4()} value={status.id}>
                        {status.name}
                      </MenuItem>
                    )
                  )}
                </CustomSelect>
              </Grid>
              <Grid item xs={6} className="residential-period">
                <CustomSelect
                  label="Years"
                  onChange={(event: any) =>
                    onChangeAddressDetail(event, "yearsAtAddress")
                  }
                  value={address.addressStatus.yearsAtAddress}
                  isRequired
                  errors={errors.residentialPeriod}
                >
                  {Array.from(Array(25).keys()).map((n: number) => (
                    <MenuItem key={v4()} value={n}>
                      {`${n} Years`}
                    </MenuItem>
                  ))}
                </CustomSelect>
              </Grid>
              <Grid item xs={6} className="residential-period">
                <CustomSelect
                  label="Months"
                  onChange={(event: any) =>
                    onChangeAddressDetail(event, "monthsAtAddress")
                  }
                  value={address.addressStatus.monthsAtAddress}
                  isRequired
                  errors={errors.residentialPeriod}
                >
                  {Array.from(Array(12).keys()).map((n: number) => (
                    <MenuItem key={v4()} value={n}>
                      {`${n} Months`}
                    </MenuItem>
                  ))}
                </CustomSelect>
              </Grid>
              <Grid item xs={12}>
                <FormControlLabel
                  control={
                    <Checkbox
                      checked={address.isPrimaryAddress}
                      onChange={(event) =>
                        onChangeAddressDetail(event, "isPrimaryAddress")
                      }
                      value="isPrimaryAddress"
                    />
                  }
                  label="This is their current address"
                />
              </Grid>
            </Grid>
            {loading && <Loading />}
          </ResponsiveDialogContentStyles>
        </ResponsiveDialog>
      )}
      {openConfirmation && (
        <ResponsiveDialog
          open={openConfirmation}
          maxWidth="xs"
          onConfirm={onDelete}
          onCancel={() => setOpenConfirmation(false)}
          showActionButtons
          okText="Yes"
          cancelText="No"
          isConfirmationDialog
        >
          <div className="confirmation-message">
            Are you sure you want to delete this address?
          </div>
        </ResponsiveDialog>
      )}
    </AddressDialogStyles>
  );
};
