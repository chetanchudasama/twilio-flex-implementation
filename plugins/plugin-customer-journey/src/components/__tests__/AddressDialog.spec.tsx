import React from "react";
import { create } from "react-test-renderer";
import { mount, shallow } from "enzyme";
import { act } from "react-dom/test-utils";
import { ResponsiveDialog, AddressItemModel } from "@common/components";
import { AddressDialog } from "../CustomerDetailAction/AddressHistory/AddressDialog/AddressDialog";
import { AddressDialogProps } from "../CustomerDetailAction/AddressHistory/AddressDialog/AddressDialog.Props";

jest.mock("@common/components", () => {
  const components = jest.requireActual("@common/components");
  return {
    ...components,
    ResponsiveDialog: jest.fn(({ children }) => children),
  };
});

jest.mock("../AddressSearch/AddressSearch", () => {
  return {
    AddressSearch: () => {
      return <div className="address-search">Address Search</div>;
    },
  };
});

const addressDetail = {
  postcode: "NN6 2XS",
  buildingName: "Grange",
  subBuilding: "Grange-1",
  buildingNumber: "110",
  streetName: "43 Buckingham Rd",
  town: "THORNBY",
  addressStatus: {
    addressStatusId: 1,
    addressStatusName: "Home Owner",
    yearsAtAddress: 1,
    monthsAtAddress: 12,
  },
  isPrimaryAddress: true,
};

const props: AddressDialogProps = {
  isEdit: true,
  open: true,
  loading: false,
  activeAddress: addressDetail,
  onDelete: jest.fn(() => {}),
  onClose: jest.fn(() => {}),
  onSave: jest.fn(() => {}),
};

describe("AddressDialog", () => {
  it("renders correctly", () => {
    const wrapper = create(<AddressDialog {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("when value of isEdit prop is false, UI should be rendered with empty fields", () => {
    props.isEdit = false;
    props.activeAddress = new AddressItemModel();
    const wrapper = create(<AddressDialog {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("should emit onClose method, when cancel button is clicked", () => {
    const wrapper = shallow(<AddressDialog {...props} />);

    const spy = jest.spyOn(props, "onClose");

    act(() => {
      wrapper.children().props().onCancel();
    });

    expect(spy).toHaveBeenCalled();
  });

  it("should render Confirmation dialog, when delete button is clicked", () => {
    const wrapper = shallow(<AddressDialog {...props} />);

    expect(wrapper.find(ResponsiveDialog)).toHaveLength(1);
    act(() => {
      wrapper.children().props().onDelete();
    });

    expect(wrapper.find(ResponsiveDialog)).toHaveLength(2);
    expect(wrapper.find(".confirmation-message").exists()).toBeTruthy();
    expect(wrapper.find(".confirmation-message").text()).toContain(
      "Are you sure you want to delete this address?"
    );
  });

  it("should close confirmation dialog, when No button is clicked from confirmation dialog", () => {
    const wrapper = shallow(<AddressDialog {...props} />);

    act(() => {
      wrapper.children().props().onDelete();
    });

    expect(wrapper.find(".confirmation-message").exists()).toBeTruthy();
    act(() => {
      wrapper.find(ResponsiveDialog).at(1).props().onCancel();
    });

    expect(wrapper.find(".confirmation-message").exists()).toBeFalsy();
  });

  it("should emit onDelete method, when Yes button is clicked from confirmation dialog", () => {
    const wrapper = shallow(<AddressDialog {...props} />);

    const spy = jest.spyOn(props, "onDelete");

    act(() => {
      wrapper.children().props().onDelete();
    });
    expect(wrapper.find(".confirmation-message").exists()).toBeTruthy();

    act(() => {
      wrapper.find(ResponsiveDialog).at(1).props().onConfirm();
    });

    expect(spy).toHaveBeenCalled();
  });

  it("should render validation message", async () => {
    props.isEdit = false;
    props.activeAddress = new AddressItemModel();
    const wrapper = mount(<AddressDialog {...props} />);

    const spy = jest.spyOn(props, "onSave");

    act(() => {
      wrapper.find(ResponsiveDialog).at(0).props().onConfirm();
    });

    expect(wrapper.find(".residential-status").at(0).html()).toContain(
      "The residential status is required."
    );
    expect(wrapper.find(".residential-period").at(0).html()).toContain(
      "Residential period is required."
    );

    expect(spy).not.toHaveBeenCalled();
  });

  it("when edit address button is clicked, onSave method should be called", async () => {
    props.isEdit = true;
    props.activeAddress = addressDetail;
    const wrapper = mount(<AddressDialog {...props} />);

    const spy = jest.spyOn(props, "onSave");

    act(() => {
      wrapper.find(ResponsiveDialog).at(0).props().onConfirm();
    });

    expect(spy).toHaveBeenCalledWith(addressDetail);
  });

  it("should render AddressSearch component correctly in UI", async () => {
    const wrapper = mount(<AddressDialog {...props} />);

    expect(wrapper.find(".address-search").exists()).toBeTruthy();
  });
});
