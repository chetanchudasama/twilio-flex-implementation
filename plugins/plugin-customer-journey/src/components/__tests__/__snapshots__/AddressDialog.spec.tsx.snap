// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressDialog renders correctly 1`] = `
<div
  className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 css-0"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 css-xv6sq1"
  >
    <div
      className="MuiGrid-container-1 MuiGrid-spacing-xs-8-23 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-space-between-20"
    >
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 address-container"
      >
        <label
          className="MuiFormLabel-root-109 MuiFormLabel-required-114 MuiInputLabel-required-102 MuiInputLabel-root-98 MuiInputLabel-animated-106 MuiInputLabel-shrink-105"
          data-shrink={true}
        >
          Address
          <span
            className="MuiFormLabel-asterisk-115"
          >
             *
          </span>
        </label>
        <div
          className="address-search"
        >
          Address Search
        </div>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 residential-status"
      >
        <div
          className="MuiFormControl-root-121 MuiFormControl-marginDense-123 MuiFormControl-fullWidth-124 CustomSelectContent-formControl-116"
        >
          <label
            className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiFormLabel-required-114 MuiInputLabel-required-102 MuiInputLabel-root-98 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105 MuiInputLabel-marginDense-104 CustomSelectContent-label-117"
            data-shrink={true}
          >
            Residential status
            <span
              className="MuiFormLabel-asterisk-115"
            >
               *
            </span>
          </label>
          <div
            className="MuiInputBase-root-145 MuiOutlinedInput-root-132 MuiInputBase-formControl-146 MuiInputBase-marginDense-152 CustomSelectContent-selectWithLabel-119 "
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-162 MuiOutlinedInput-notchedOutline-139"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-163"
                style={
                  Object {
                    "width": 0,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <div
              className="MuiSelect-root-125"
            >
              <div
                aria-haspopup="true"
                aria-pressed="false"
                className="MuiSelect-select-126 MuiSelect-selectMenu-129 MuiInputBase-input-155 MuiOutlinedInput-input-140 MuiInputBase-inputMarginDense-156 MuiOutlinedInput-inputMarginDense-141"
                id="select-Residential status"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                Home Owner
              </div>
              <input
                name="Residential status"
                type="hidden"
                value={1}
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-164 MuiSelect-icon-131"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-6-35 residential-period"
      >
        <div
          className="MuiFormControl-root-121 MuiFormControl-marginDense-123 MuiFormControl-fullWidth-124 CustomSelectContent-formControl-116"
        >
          <label
            className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiFormLabel-required-114 MuiInputLabel-required-102 MuiInputLabel-root-98 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105 MuiInputLabel-marginDense-104 CustomSelectContent-label-117"
            data-shrink={true}
          >
            Years
            <span
              className="MuiFormLabel-asterisk-115"
            >
               *
            </span>
          </label>
          <div
            className="MuiInputBase-root-145 MuiOutlinedInput-root-132 MuiInputBase-formControl-146 MuiInputBase-marginDense-152 CustomSelectContent-selectWithLabel-119 "
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-162 MuiOutlinedInput-notchedOutline-139"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-163"
                style={
                  Object {
                    "width": 0,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <div
              className="MuiSelect-root-125"
            >
              <div
                aria-haspopup="true"
                aria-pressed="false"
                className="MuiSelect-select-126 MuiSelect-selectMenu-129 MuiInputBase-input-155 MuiOutlinedInput-input-140 MuiInputBase-inputMarginDense-156 MuiOutlinedInput-inputMarginDense-141"
                id="select-Years"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                1 Years
              </div>
              <input
                name="Years"
                type="hidden"
                value={1}
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-164 MuiSelect-icon-131"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-6-35 residential-period"
      >
        <div
          className="MuiFormControl-root-121 MuiFormControl-marginDense-123 MuiFormControl-fullWidth-124 CustomSelectContent-formControl-116"
        >
          <label
            className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiFormLabel-required-114 MuiInputLabel-required-102 MuiInputLabel-root-98 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105 MuiInputLabel-marginDense-104 CustomSelectContent-label-117"
            data-shrink={true}
          >
            Months
            <span
              className="MuiFormLabel-asterisk-115"
            >
               *
            </span>
          </label>
          <div
            className="MuiInputBase-root-145 MuiOutlinedInput-root-132 MuiInputBase-formControl-146 MuiInputBase-marginDense-152 CustomSelectContent-selectWithLabel-119 "
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-162 MuiOutlinedInput-notchedOutline-139"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-163"
                style={
                  Object {
                    "width": 0,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <div
              className="MuiSelect-root-125"
            >
              <div
                aria-haspopup="true"
                aria-pressed="false"
                className="MuiSelect-select-126 MuiSelect-selectMenu-129 MuiInputBase-input-155 MuiOutlinedInput-input-140 MuiInputBase-inputMarginDense-156 MuiOutlinedInput-inputMarginDense-141"
                id="select-Months"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </div>
              <input
                name="Months"
                type="hidden"
                value={12}
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-164 MuiSelect-icon-131"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
      >
        <label
          className="MuiFormControlLabel-root-177"
        >
          <span
            className="MuiButtonBase-root-199 MuiIconButton-root-193 MuiPrivateSwitchBase-root-189 MuiCheckbox-root-183 MuiCheckbox-colorSecondary-188 MuiPrivateSwitchBase-checked-190 MuiCheckbox-checked-184"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label-198"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-164"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              <input
                checked={true}
                className="MuiPrivateSwitchBase-input-192"
                data-indeterminate={false}
                onChange={[Function]}
                type="checkbox"
                value="isPrimaryAddress"
              />
            </span>
            <span
              className="MuiTouchRipple-root-238"
            />
          </span>
          <span
            className="MuiTypography-root-202 MuiTypography-body1-211 MuiFormControlLabel-label-182"
          >
            This is their current address
          </span>
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`AddressDialog when value of isEdit prop is false, UI should be rendered with empty fields 1`] = `
<div
  className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 css-0"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 css-xv6sq1"
  >
    <div
      className="MuiGrid-container-1 MuiGrid-spacing-xs-8-23 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-space-between-20"
    >
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 address-container"
      >
        <label
          className="MuiFormLabel-root-109 MuiFormLabel-required-114 MuiInputLabel-required-102 MuiInputLabel-root-98 MuiInputLabel-animated-106 MuiInputLabel-shrink-105"
          data-shrink={true}
        >
          Address
          <span
            className="MuiFormLabel-asterisk-115"
          >
             *
          </span>
        </label>
        <div
          className="address-search"
        >
          Address Search
        </div>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 residential-status"
      >
        <div
          className="MuiFormControl-root-121 MuiFormControl-marginDense-123 MuiFormControl-fullWidth-124 CustomSelectContent-formControl-116"
        >
          <label
            className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiFormLabel-required-114 MuiInputLabel-required-102 MuiInputLabel-root-98 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105 MuiInputLabel-marginDense-104 CustomSelectContent-label-117"
            data-shrink={true}
          >
            Residential status
            <span
              className="MuiFormLabel-asterisk-115"
            >
               *
            </span>
          </label>
          <div
            className="MuiInputBase-root-145 MuiOutlinedInput-root-132 MuiInputBase-formControl-146 MuiInputBase-marginDense-152 CustomSelectContent-selectWithLabel-119 "
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-162 MuiOutlinedInput-notchedOutline-139"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-163"
                style={
                  Object {
                    "width": 0,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <div
              className="MuiSelect-root-125"
            >
              <div
                aria-haspopup="true"
                aria-pressed="false"
                className="MuiSelect-select-126 MuiSelect-selectMenu-129 MuiInputBase-input-155 MuiOutlinedInput-input-140 MuiInputBase-inputMarginDense-156 MuiOutlinedInput-inputMarginDense-141"
                id="select-Residential status"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <em>
                  Select a status
                </em>
              </div>
              <input
                name="Residential status"
                type="hidden"
                value={-1}
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-164 MuiSelect-icon-131"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-6-35 residential-period"
      >
        <div
          className="MuiFormControl-root-121 MuiFormControl-marginDense-123 MuiFormControl-fullWidth-124 CustomSelectContent-formControl-116"
        >
          <label
            className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiFormLabel-required-114 MuiInputLabel-required-102 MuiInputLabel-root-98 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105 MuiInputLabel-marginDense-104 CustomSelectContent-label-117"
            data-shrink={true}
          >
            Years
            <span
              className="MuiFormLabel-asterisk-115"
            >
               *
            </span>
          </label>
          <div
            className="MuiInputBase-root-145 MuiOutlinedInput-root-132 MuiInputBase-formControl-146 MuiInputBase-marginDense-152 CustomSelectContent-selectWithLabel-119 "
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-162 MuiOutlinedInput-notchedOutline-139"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-163"
                style={
                  Object {
                    "width": 0,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <div
              className="MuiSelect-root-125"
            >
              <div
                aria-haspopup="true"
                aria-pressed="false"
                className="MuiSelect-select-126 MuiSelect-selectMenu-129 MuiInputBase-input-155 MuiOutlinedInput-input-140 MuiInputBase-inputMarginDense-156 MuiOutlinedInput-inputMarginDense-141"
                id="select-Years"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                0 Years
              </div>
              <input
                name="Years"
                type="hidden"
                value={0}
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-164 MuiSelect-icon-131"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-6-35 residential-period"
      >
        <div
          className="MuiFormControl-root-121 MuiFormControl-marginDense-123 MuiFormControl-fullWidth-124 CustomSelectContent-formControl-116"
        >
          <label
            className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiFormLabel-required-114 MuiInputLabel-required-102 MuiInputLabel-root-98 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105 MuiInputLabel-marginDense-104 CustomSelectContent-label-117"
            data-shrink={true}
          >
            Months
            <span
              className="MuiFormLabel-asterisk-115"
            >
               *
            </span>
          </label>
          <div
            className="MuiInputBase-root-145 MuiOutlinedInput-root-132 MuiInputBase-formControl-146 MuiInputBase-marginDense-152 CustomSelectContent-selectWithLabel-119 "
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-162 MuiOutlinedInput-notchedOutline-139"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-163"
                style={
                  Object {
                    "width": 0,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <div
              className="MuiSelect-root-125"
            >
              <div
                aria-haspopup="true"
                aria-pressed="false"
                className="MuiSelect-select-126 MuiSelect-selectMenu-129 MuiInputBase-input-155 MuiOutlinedInput-input-140 MuiInputBase-inputMarginDense-156 MuiOutlinedInput-inputMarginDense-141"
                id="select-Months"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                0 Months
              </div>
              <input
                name="Months"
                type="hidden"
                value={0}
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-164 MuiSelect-icon-131"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
      >
        <label
          className="MuiFormControlLabel-root-177"
        >
          <span
            className="MuiButtonBase-root-199 MuiIconButton-root-193 MuiPrivateSwitchBase-root-189 MuiCheckbox-root-183 MuiCheckbox-colorSecondary-188"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label-198"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-164"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <input
                checked={false}
                className="MuiPrivateSwitchBase-input-192"
                data-indeterminate={false}
                onChange={[Function]}
                type="checkbox"
                value="isPrimaryAddress"
              />
            </span>
            <span
              className="MuiTouchRipple-root-238"
            />
          </span>
          <span
            className="MuiTypography-root-202 MuiTypography-body1-211 MuiFormControlLabel-label-182"
          >
            This is their current address
          </span>
        </label>
      </div>
    </div>
  </div>
</div>
`;
